pennylane/measurements.py

```diff
@@ -156,6 +136,43 @@ def __init__(
         # Queue the measurement process
         self.queue()
 
+    @property
+    @functools.lru_cache()
+    def numeric_type(self):
+        """The Python numeric type of the measurement result.
+
+        Returns:
+            tuple: The output numeric type; ``int``, ``float`` or ``complex``.
+
+        Raises:
+            QuantumFunctionError: the return type of the measurement process is
+                unrecognized and cannot deduce the numeric type
+        """
+        if self.return_type in (Expectation, Variance, Probability):
```

Comment:
This PR could affect directly the creation of `VnEntropy` and `MutualInfo`, have you seen potential blockers by reviewing mutual info?

Reply:
No blockers, likely this logic just has to be extended.